baybe/constraints/continuous.py

```diff
@@ -31,3 +36,87 @@ class ContinuousLinearInequalityConstraint(ContinuousConstraint):
     The class has no real content as it only serves the purpose of
     distinguishing the constraints.
     """
+
+
+@define
+class ContinuousCardinalityConstraint(ContinuousConstraint):
+    """Class for continuous cardinality constraints.
+
+    Places a constraint on the set of nonzero (i.e. "active") values among the
+    specified parameter, bounding it between the two given integers,
+        ``min_cardinality`` <= | {1(p_i != 0)}_i | <= ``max_cardinality``
+    where ``1`` is the Kronecker delta function and ``{p_i}`` are the parameters
+    specified for the constraint.
+
+    Note that this can be equivalently regarded a L0-constraint on the vector containing
+    the specified parameters.
+    """
+
+    min_cardinality: int = field(default=0, validator=[instance_of(int), ge(0)])
```

Comment:
shouldnt these names be consistent with the names we chose for `Searchspace.discrete.from_simplex` (`min_nonzero` and `max_nonzero` in there) ? not saying these are wrong but they shiuld be same in both places

Reply:
as agreed, we'll keep things as they are for now and then kick the `min_nonzero`/`max_nonzero` arguments from the `from_simplex` constructor altogether, replacing them with the `DiscreteCardinalityConstraint`, that can be directly added to the constraints list.